src/modules/battery_status/battery_status.cpp

```diff
@@ -280,9 +281,7 @@ BatteryStatus::task_spawn(int argc, char *argv[])
 bool
 BatteryStatus::init()
 {
-	ScheduleOnInterval(SAMPLE_INTERVAL_US);
-
-	return true;
+	return _adc_report_sub.registerCallback();;
```

Comment:
We need to be careful that the pub freq matches the configured filter freq.
```suggestion
	return _adc_report_sub.registerCallback();
```

Otherwise looks good, there's still a compile error though.